### ISSUE TITLE
'addAgeTemp' error in efficient branch

### ISSUE THREAD

**Issue by chandryou** [OHDSI/PatientLevelPrediction#120]
I cannot run the package with adding age covariates in temporal analysis, through efficient branch. 

I think there are multiple problems. 

1. If I add Age in temporal covariate extraction, then it doesn't have timeId. So tidycovariatedata function cannot work on it. 

2. 
In efficient branch, addAgeTemp function (Formatting.r, line #619) doesn't seem working for me. 

Can I modify featureextraction package rather than patientlevelpredication package for this problem? If 
@schuemie and @jreps allow, I want to do so.

**Comment by jreps** [comment]
Hi Chan,  what age covariates are you adding?  When I wrote the code the temporal feature extraction was extracting the rowId, covariateId, covariateValue and timeId - so to PLP code searched for the age covariateId and created the relevant covariateValues for each timeId, for the non-age covariates the covariateValues stay the same across the timeIds.  There may be some bugs when continuous age isn't in the plpData I can fix.  I'll need to check whether feature extraction has changed the temporal covariate data as that might also be causing issues.

The tidycovariatedata should work for temporal data - I might need to check for recent edits and fix any issues when applying the models though.

**Comment by jreps** [comment]
Hi Chan,  

I was able to download data with the latest FeatureExtraction and fixed the issues.  There are two issues:

1. In PLP we have removeRedundancy = T for FeatureExtraction::tidyCovariateData() and this was causing an odd empty ff issue.  I decided to just set that the F if the data are temporal.
2. The timeId for non-temporal covariates used to be 0, so the code was written for this but it seems to be NA now, which was causing issues.  I've updated the code to assume the non-temporal data have NA for the timeID

My code is now running.  I've committed these edits to the efficient branch. After a little more testing of the efficient branch I want to merge it into master.  The other python classifiers should be a lot quicker in efficient (randomForest, adaBoost, decisionTree, naiveBayes, mlp) and they use the R library reticulate instead of PythonInR.

Best wishes,
Jenna
